templates/todo/api/csharp-cosmos-sql/Program.cs

```diff
@@ -15,8 +15,7 @@
     }
 }));
 builder.Services.AddControllers();
-var options = new ApplicationInsightsServiceOptions { ConnectionString = builder.Configuration["APPLICATIONINSIGHTS_CONNECTION_STRING"] };
-builder.Services.AddApplicationInsightsTelemetry(options);
+builder.Services.AddApplicationInsightsTelemetry(builder.Configuration);
```

Comment:
.NET application Insights reads `APPLICATIONINSIGHTS_CONNECTION_STRING` env var from configuration/env on its own and supports binding to configuration, nothing extra is needed.

It also supports missing configuration and connection string writing to debug output or integrating with VS without ikey.